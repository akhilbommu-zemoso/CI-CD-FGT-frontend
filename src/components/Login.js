import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import TrendingUpSharpIcon from "@material-ui/icons/TrendingUpSharp";
import FTButton from "../atoms/FTButton";
import { FTCard } from "../atoms/FTCard";
const useStyles = makeStyles({
    card: {
        borderRadius: "10px",
        width: "400px",
        height: "350px",
        marginTop: "100px",
    },
    centered: {
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        textAlign: "center",
    },
    header: {
        fontSize: 30,
        color: "black",
        fontWeight: 900,
        marginTop: "20px",
    },
    logo: {
        width: "70px",
        height: "70px",
        fill: "white",
        backgroundColor: "rgb(4,136,255)",
        borderRadius: "50%",
        padding: "5px",
    },
    content: {
        marginTop: "20px",
    },
    blueText: {
        color: "blue",
        fontWeight: 900,
    },
    footer: {
        marginTop: "30px",
    },
    color: {
        marginTop: "30px",
        color: "white",
        backgroundColor: "rgb(4,136,255)",
        "&:hover": {
            color: "white",
            backgroundColor: "rgb(4,136,255)",
        },
    },
});

export default function Login() {
    const authLogin = () => {
        var domain = process.env.REACT_APP_AUTH0_DOMAIN;
        var client_id = process.env.REACT_APP_AUTH0_CLIENTID;
        var redirect_uri = window.location.origin+""+process.env.REACT_APP_AUTH0_REDIRECT_URI;
        console.log(redirect_uri)
        const url = `https://${domain}/authorize?scope=openid%20profile%20email%20user_metadata%20picture&audience=login&response_type=token id_token&client_id=${client_id}&redirect_uri=${redirect_uri}&nonce=test&connection=google-oauth2`;
        window.location = url;
    };
    const classes = useStyles();
    const header = () => {
        return (
            <>
                <TrendingUpSharpIcon fontSize="large" className={classes.logo} />
                <Typography className={classes.header}>
                    Please Click on Login
                </Typography>
            </>
        );
    };
    return (
        <div className={classes.centered}>
            <FTCard
                cardStyle={classes.card}
                headerStyle={classes.header}
                contentStyle={classes.content}
                footerStyle={classes.footer}
                header={header()}
                content={
                    <>
                        to continue
                        <span className={classes.blueText}> to ZE-Transaction</span>
                    </>
                }
                footer={
                    <FTButton data-testid="button" name="Login" method={authLogin} />
                }
            />
        </div>
    );
}
