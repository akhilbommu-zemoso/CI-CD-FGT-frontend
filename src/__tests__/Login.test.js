import React from 'react'
import { render, cleanup, fireEvent, within, getAllByText, getByText } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import Login from '../components/Login'
afterEach(cleanup)

it('should take a snapshot', () => {
    const { asFragment, getByTestId } = render(<Router><Login /></Router>)
    expect(asFragment(<Router><Login /></Router>)).toMatchSnapshot()

});
it("Checks for button", () => { 
    const { getByTestId } = render(<Router><Login /></Router>)
    const button = getByTestId("button")
    expect(button).toBeInTheDocument
})

it("checks for header", () => {
    const { getByTestId, getByText } = render(<Router><Login /></Router>)
    const heading = getByText("Please Click on Login")
    expect(heading).toBeInTheDocument
})
const authLogin = jest.fn();
it("handlesubmit event should work ", async () => {
    const { getByTestId } = render(<Login />);
    fireEvent.click(getByTestId("button"));
    expect(authLogin).toBeCalled
});

