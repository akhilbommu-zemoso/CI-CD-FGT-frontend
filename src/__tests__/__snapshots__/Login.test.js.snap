// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-centered-2"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-header-3 MuiTypography-body1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-logo-4 MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6h-6z"
            />
          </svg>
        </p>
        <p
          class="MuiTypography-root makeStyles-header-3 MuiTypography-body1"
        >
          Please Click on Login
        </p>
        <p />
        <p
          class="MuiTypography-root makeStyles-content-5 MuiTypography-body1"
        >
          to continue
          <span
            class="makeStyles-blueText-6"
          >
             to ZE-Transaction
          </span>
        </p>
        <p
          class="MuiTypography-root makeStyles-footer-7 MuiTypography-body1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-color-78 MuiButton-contained"
            data-testid="button"
            id="button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Login
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
